pkg/workload/workload.go

```diff
@@ -170,9 +170,11 @@ func (psr *PodSetResources) ScaledTo(newCount int32) *PodSetResources {
 		Flavors:  maps.Clone(psr.Flavors),
 	}
 
-	scaleDown(ret.Requests, int64(ret.Count))
-	scaleUp(ret.Requests, int64(newCount))
-	ret.Count = newCount
+	if psr.Count != 0 {
```

Comment:
```suggestion
	if psr.Count == 0 || psr.Count == newCount {
	  return psr
	}
```
is it safe to do this guard clause (which would be placed at the top of the function, before the copy)? i.e. do any callers modify this object, outside of ScaledTo?